agent/getIn.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/**
+ * Retrieves the value from the path of nested objects
+ * @param  {Object} base Base or root object for path
+ * @param  {Array<String>} path nested path
+ * @return {any}      Value at end of path or `mull`
+ */
+function getIn(base, path) {
+  return path.reduce((obj, attr) => {
+    if (obj) {
+      if (obj.hasOwnProperty(attr)) {
+        return obj[attr];
+      }
+      if (typeof obj[Symbol.iterator] === 'function') {
+        // Convert iterable to array and return array[index]
+        return [...obj][attr];
```

Comment:
1. `getIn` was duplicated code moved fro Bridge.js#441 and Agent.js#401 and updated to support iterators. The complexity of the previous implementation hasn't changed (with the exception of spreading the iterable on line 25).

2. This does work with key-value maps (if they are iterable) with one caveat, in order to support maps where the key is not a primitive the tool must treat the `[key, value]` entry as an iterable as well resulting in something like this:

```
myMap: Map(..)
  0: [
    0: <complex key A>
    1:  "some value"
  ]
  1: [
    0: "string key B"
    1:  "some other value"
  ]
  ...
```

see **Improvement of Keyed iterables/iterators** in this [comment](https://github.com/facebook/react-devtools/pull/459#issue-192774346)

I hope I'm making sense here.